src/dispatch/database/base.py

```diff
@@ -0,0 +1,66 @@
+import re
```

Comment:
nit: we seem to be using "base" a lot e.g. database.py import Base class now it's database.base import Base, this module might better named "core" or similar.

Reply:
Yeah, sure. Honestly, I did not spend a lot of time thinking about it.